File: csharp/0300最长递增子序列.cs

public partial class Solution
{
    public int LengthOfLIS(int[] nums)
    {
        int res = 1;
        int size = nums.Length;

        int[] dp = new int[size];
        for (int i = 0; i < size; i++)
            dp[i] = 1;

        for (int i = 1; i < size; i++)
        {
            for (int j = 0; j < i; j++)
            {
                if (nums[i] > nums[j])
                    dp[i] = Math.Max(dp[i], dp[j] + 1);
            }

            res = Math.Max(res, dp[i]);
        }

        return res;
    }
}

File: python/0300最长递增子序列.py

# dp，注意不是最长连续递增
def lengthOfLIS(nums: list) -> int:
    size=len(nums)

    dp=[1]*size

    for i in range(1,size):

        for j in range(i):
            if(nums[i]>nums[j]):
                dp[i]=max(dp[i],dp[j]+1)
    
    return max(dp)
print(lengthOfLIS([1, 3, 6, 7, 9, 4, 10, 5, 6]))
